integrate cloud social provider (to the dev build)

trevj opened the pull request:
This adds the Cloud social provider, to the dev build only.

As discussed, login doesn't make a whole lot of sense right now. However, it will help some of the other efforts underway to have some minimal cloud instance support in the client.

This depends on token support, under review here:
https://github.com/uProxy/uproxy-lib/pull/314

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy/2040)
<!-- Reviewable:end -->

dborkan commented:
Some small items, then :+1: 
<!-- Reviewable comment -K2nA6ps1CjQOFSorNOo:118217815 -->

---

Review status: 0 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

<sup>**[src/generic_core/globals.ts, line 42 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2040#-K2n9jpM65Rrs1ZlKBK5)** ([raw file](https://github.com/uproxy/uproxy/blob/506eb5b8cd42370edc572c87a662109eb519a55e/src/generic_core/globals.ts#L42)):</sup>
Is it intentional to check in debug logging?  I'm fine with it, but not sure if might slow things down a little or annoy other developers (the social code currently prints a ton of debug trace...which is useful for debugging but otherwise some people find it annoying)
<!-- Reviewable comment -K2n9jpM65Rrs1ZlKBK6:-1938032175 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 129 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2040#-K2n9PGUSS8BqfQlhpaY)** ([raw file](https://github.com/uproxy/uproxy/blob/506eb5b8cd42370edc572c87a662109eb519a55e/src/generic_ui/polymer/invite-user.html#L129)):</sup>
I don't think you need to hide this for cloud: the accept-user-invite page actually works the same regardless of which network is selected in invite-user.html (the network name is encoded in the invite token/url so the user doesn't need to pick).  This is a little bit weird UX right now but will be better after Lucy's changes to move the invite/login screen
<!-- Reviewable comment -K2n9PGUSS8BqfQlhpaZ:2075147741 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2040)
<!-- Sent from Reviewable.io -->

trevj commented:
Review status: 0 of 8 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 129 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2040#-K2n9PGUSS8BqfQlhpaY)** ([raw file](https://github.com/uproxy/uproxy/blob/506eb5b8cd42370edc572c87a662109eb519a55e/src/generic_ui/polymer/invite-user.html#L129)):</sup>
Ah! Makes sense. Removed, thanks.
<!-- Reviewable comment -K2s5sYhtGhW2zX3lsAx:-1406994453 -->

---

<sup>**[src/generic_core/globals.ts, line 42 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2040#-K2n9jpM65Rrs1ZlKBK5)** ([raw file](https://github.com/uproxy/uproxy/blob/506eb5b8cd42370edc572c87a662109eb519a55e/src/generic_core/globals.ts#L42)):</sup>
Oops! I had done this just for testing. Reverted, thanks.
<!-- Reviewable comment -K2s5wSK6FDdyofEDHAU:520303008 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2040)
<!-- Sent from Reviewable.io -->
